source/jormungandr/jormungandr/street_network/kraken.py

```diff
@@ -139,8 +139,21 @@ def _direct_path(
         response = instance.send_and_receive(req, request_id=request_id)
         if should_invert_journey:
             return self._reverse_journeys(response)
+
+        def has_bss_section(journey):
+            # Here is a little trick with python's generator
+            # the next 3 lines check not only the existences of BSS_RENT and BSS_PUT_BACK, but also check the fact that
+            # BSS_RENT must be located Before BSS_PUT_BACK
```

Comment:
Where is hidden the magic that check the BSS_RENT is before BSS_PUT_BACK ?

Reply:
The magic comes from the `next` function, that will iterate the sections until it finds a bss_rent.
Then iterate over what's left of the generator to find a bass_put_back.

Reply:
But here the generator is not kept between bss_rent and bss_put_back, right ?

Reply:
But it is between the 2 calls to `next` :) 
The generator `sections` can be seen as an iterator if you prefer.

Reply:
Ok, I get it ! :+1:

Reply:
@pbench 
```python
l = [1,2,3,4,5,6]
magic_generator = (x for x in l)
should_be_true  = next( (True for x in magic_generator if x == 3 ), False)
should_be_false = next( (True for x in magic_generator if x == 2 ), False)

```